src/dashboard.js

```diff
@@ -127,6 +131,25 @@ class Dashboard extends Component {
     return [start, end];
   }
 
+  processDeviceConnStatus = (item) =>{
+    try{
+      let newRec = {
+        time: Date.parse(item.time),
+        isConn: item.operationName === 'deviceConnect'
+      };
+      let deviceId = JSON.parse(item.properties).deviceId;
+      let deviceConns = this.connRecords.get(deviceId);
+      if(!deviceConns){
+        deviceConns = [];
+        this.connRecords.set(deviceId, deviceConns);
+      }
+      if(!deviceConns.find(rec => rec.time == newRec.time)){
+        deviceConns.push(newRec);
+      }
+    }catch(e){
+    }

```

Comment:
Is this catch for JSON parse exception? Better to at least console.log something about the error here.